doc/common-illnesses.rst

```diff
@@ -0,0 +1,176 @@
+To check the names of these modules, see the list in the `Python standard library documentation`_.
+
+.. _Python standard library documentation: https://docs.python.org/3/library/
+
+
+
+===========================
+Completion/help not working
+===========================
+
+To provide code completions, help and real-time analysis in the Editor, Spyder uses the Python Language Server (PyLS), an implementation of the Language Server Protocol specification used by VSCode, Atom and other popular editors/IDEs.
+Most help and completion issues lie outside of Spyder's control, and are either limitations with PyLS or the code that is being introspected, but some can be worked around.
+
+
+Object missing docstring
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If nothing is displayed in the calltip, hover hint or help pane, the object you're trying to introspect may not have a docstring.
+
+.. image:: images/common-illnesses/common-illnesses-missing-docstring.png
+   :alt: Docstring not found in help pane
+   :width: 500px
+
+In this case, the only solution is to add one in the source code of the original function, method or class.
+
+
+Object cannot be found
+~~~~~~~~~~~~~~~~~~~~~~
+
+Some objects, whether due to being written in C, Cython or another language; generated dynamically at runtime; or being a method of an object you create, cannot be easily found without executing the code.
+
+.. image:: images/common-illnesses/common-illnesses-not-found.png
+   :alt: Object not found in help pane
+   :width: 500px
+
+However, once you run your code in the :doc:`ipythonconsole`, you might be able to get help and completions on the object there.
+
+
+LSP has stopped working
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Occasionally, especially after using Spyder for a while, code completion, help and analysis may stop working.
+If this is the case, you can check LSP status with the :guilabel:`LSP Python` item in Spyder's status bar at the bottom of the screen, and restart it by right-clicking it and selecting the :guilabel:`Restart Python Language Server` item.
+
+.. image:: images/common-illnesses/common-illnesses-LSP-restart.png
+   :alt: Spyder with LSP restart dialog
+
+
+Spyder bug/dependency issue
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Given the variety of dependencies involved in making LSP work, an incompatible or out of date version in your environment can result in error messages, incomplete results, or help/analysis not working at all.
+
+To address this, first try updating Anaconda and Spyder as described in :doc:`basic-first-aid`.
+If the issue still isn't resolved, update the various relevant dependencies with:
+
+.. code-block:: bash
+
+   conda update python-language-server
+
+
+
+===============
+Plugin Problems
+===============
+
+Plugin does not work at all
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you have installed a Spyder plug-in, but Spyder does not seem to see it, then first select the :guilabel:`Dependencies` item under the :guilabel:`Help` menu.
```

Comment:
```suggestion
If you have installed a Spyder plugin, but Spyder does not seem to see it, first select the :guilabel:`Dependencies` item under the :guilabel:`Help` menu.
```

Improve flow

Reply:
```suggestion
If you have installed a Spyder plugin, but you can't see it, go to the :guilabel:`Panes` submenu of the :guilabel:`View` menu and select the plugin's name, which should make its pane visible.
If you don't see the plugin there, select the :guilabel:`Dependencies` item under the :guilabel:`Help` menu and see if the plugin appears at the bottom.
```

Flip the order of the deps dialog and view menu, since its faster for users if the plugin is already working, and simplify the prose.